### PR TITLE
Added a generic function to validate input, and added support for failure case in RPC and client

### PR DESCRIPTION
**Pull request by grandpharaon**
Added a generic function to validate input, and added support for failure case in RPC and client.

The validation function can be used by anyone for their functions. You need to pass in the input string and a set<char> containing the valid characters allowed in your expression. The function returns true if all chars in string are in the set, otherwise returns false.

### CODE REVIEW

**Review by jpratheeppornsak — APPROVED** [review]

File: server/Calculator.cpp [hunk 1]
```diff
@@ -4,19 +4,29 @@
...
 
-
 using namespace std;
 
 string Calculator::calculateExpression(string inExpr)
 {
-   //Parse input expression into tokens using spaces and operators as delimiters
-   vector<string> expTokens = expTokenize(inExpr);
+   //Validate input expression (i.e. containing valid characters)
+   set<char> validChars = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
```
Comment (troy8chen) [inline server/Calculator.cpp:12]: Should we also include A, B, C, D, E, F for the hex expression for the validChars as well?

Comment (grandpharaon) [reply server/Calculator.cpp:12]: @troy8chen, this is the set for calculate expression function only. For your purposes, you will create a similar set with the chars that are valid for your specific function and pass it in as a parameter to the validate function

**Review by chancerh — APPROVED** [review]
Thanks!

**Comment by grandpharaon** [comment]
Thanks for reviewing... Merging

**Pull request merged**